{"nbformat":4,"nbformat_minor":0,"metadata":{"accelerator":"GPU","colab":{"name":"first_trainer_cls.ipynb","provenance":[],"collapsed_sections":[],"toc_visible":true,"machine_shape":"hm","mount_file_id":"1zPvqjKhmzOvvkV_-PFpX3gaSeS1lP0wI","authorship_tag":"ABX9TyPVxxM6Qu2ZCd7WEikdMRuw"},"kernelspec":{"name":"python3","display_name":"Python 3"},"language_info":{"codemirror_mode":{"name":"ipython","version":3},"file_extension":".py","mimetype":"text/x-python","name":"python","nbconvert_exporter":"python","pygments_lexer":"ipython3","version":"3.7.3"}},"cells":[{"cell_type":"code","metadata":{"colab_type":"code","executionInfo":{"status":"ok","timestamp":1589978167405,"user_tz":-120,"elapsed":5144,"user":{"displayName":"horst hammer","photoUrl":"","userId":"16387219538505432860"}},"id":"spDiWJ97I0vZ","outputId":"b21f6d19-d6d8-4657-e938-c21091150454","colab":{"base_uri":"https://localhost:8080/","height":311}},"source":["# %cd ..\n","!nvidia-smi"],"execution_count":5,"outputs":[{"output_type":"stream","text":["Wed May 20 12:36:05 2020       \n","+-----------------------------------------------------------------------------+\n","| NVIDIA-SMI 440.82       Driver Version: 418.67       CUDA Version: 10.1     |\n","|-------------------------------+----------------------+----------------------+\n","| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |\n","| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |\n","|===============================+======================+======================|\n","|   0  Tesla P100-PCIE...  Off  | 00000000:00:04.0 Off |                    0 |\n","| N/A   33C    P0    25W / 250W |      0MiB / 16280MiB |      0%      Default |\n","+-------------------------------+----------------------+----------------------+\n","                                                                               \n","+-----------------------------------------------------------------------------+\n","| Processes:                                                       GPU Memory |\n","|  GPU       PID   Type   Process name                             Usage      |\n","|=============================================================================|\n","|  No running processes found                                                 |\n","+-----------------------------------------------------------------------------+\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"colab_type":"code","id":"j5KSeQKwuZYB","outputId":"0e2c2abc-4b85-48b0-f836-a76a501ec8f2","executionInfo":{"status":"ok","timestamp":1589978302419,"user_tz":-120,"elapsed":139803,"user":{"displayName":"horst hammer","photoUrl":"","userId":"16387219538505432860"}},"colab":{"base_uri":"https://localhost:8080/","height":1000}},"source":["!pip install gdown\n","!pip install tensorflow-gpu --user\n","!pip install --upgrade keras\n","!pip install ann_visualizer\n","!pip install graphviz"],"execution_count":6,"outputs":[{"output_type":"stream","text":["Requirement already satisfied: gdown in /usr/local/lib/python3.6/dist-packages (3.6.4)\n","Requirement already satisfied: requests in /usr/local/lib/python3.6/dist-packages (from gdown) (2.23.0)\n","Requirement already satisfied: six in /usr/local/lib/python3.6/dist-packages (from gdown) (1.12.0)\n","Requirement already satisfied: tqdm in /usr/local/lib/python3.6/dist-packages (from gdown) (4.41.1)\n","Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.6/dist-packages (from requests->gdown) (2020.4.5.1)\n","Requirement already satisfied: chardet<4,>=3.0.2 in /usr/local/lib/python3.6/dist-packages (from requests->gdown) (3.0.4)\n","Requirement already satisfied: idna<3,>=2.5 in /usr/local/lib/python3.6/dist-packages (from requests->gdown) (2.9)\n","Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in /usr/local/lib/python3.6/dist-packages (from requests->gdown) (1.24.3)\n","Collecting tensorflow-gpu\n","\u001b[?25l  Downloading https://files.pythonhosted.org/packages/31/bf/c28971266ca854a64f4b26f07c4112ddd61f30b4d1f18108b954a746f8ea/tensorflow_gpu-2.2.0-cp36-cp36m-manylinux2010_x86_64.whl (516.2MB)\n","\u001b[K     |████████████████████████████████| 516.2MB 28kB/s \n","\u001b[?25hRequirement already satisfied: astunparse==1.6.3 in /usr/local/lib/python3.6/dist-packages (from tensorflow-gpu) (1.6.3)\n","Requirement already satisfied: h5py<2.11.0,>=2.10.0 in /usr/local/lib/python3.6/dist-packages (from tensorflow-gpu) (2.10.0)\n","Requirement already satisfied: google-pasta>=0.1.8 in /usr/local/lib/python3.6/dist-packages (from tensorflow-gpu) (0.2.0)\n","Requirement already satisfied: opt-einsum>=2.3.2 in /usr/local/lib/python3.6/dist-packages (from tensorflow-gpu) (3.2.1)\n","Requirement already satisfied: protobuf>=3.8.0 in /usr/local/lib/python3.6/dist-packages (from tensorflow-gpu) (3.10.0)\n","Requirement already satisfied: absl-py>=0.7.0 in /usr/local/lib/python3.6/dist-packages (from tensorflow-gpu) (0.9.0)\n","Requirement already satisfied: tensorboard<2.3.0,>=2.2.0 in /usr/local/lib/python3.6/dist-packages (from tensorflow-gpu) (2.2.1)\n","Requirement already satisfied: numpy<2.0,>=1.16.0 in /usr/local/lib/python3.6/dist-packages (from tensorflow-gpu) (1.18.4)\n","Requirement already satisfied: termcolor>=1.1.0 in /usr/local/lib/python3.6/dist-packages (from tensorflow-gpu) (1.1.0)\n","Requirement already satisfied: wrapt>=1.11.1 in /usr/local/lib/python3.6/dist-packages (from tensorflow-gpu) (1.12.1)\n","Requirement already satisfied: gast==0.3.3 in /usr/local/lib/python3.6/dist-packages (from tensorflow-gpu) (0.3.3)\n","Requirement already satisfied: keras-preprocessing>=1.1.0 in /usr/local/lib/python3.6/dist-packages (from tensorflow-gpu) (1.1.2)\n","Requirement already satisfied: wheel>=0.26; python_version >= \"3\" in /usr/local/lib/python3.6/dist-packages (from tensorflow-gpu) (0.34.2)\n","Requirement already satisfied: tensorflow-estimator<2.3.0,>=2.2.0 in /usr/local/lib/python3.6/dist-packages (from tensorflow-gpu) (2.2.0)\n","Requirement already satisfied: scipy==1.4.1; python_version >= \"3\" in /usr/local/lib/python3.6/dist-packages (from tensorflow-gpu) (1.4.1)\n","Requirement already satisfied: six>=1.12.0 in /usr/local/lib/python3.6/dist-packages (from tensorflow-gpu) (1.12.0)\n","Requirement already satisfied: grpcio>=1.8.6 in /usr/local/lib/python3.6/dist-packages (from tensorflow-gpu) (1.29.0)\n","Requirement already satisfied: setuptools in /usr/local/lib/python3.6/dist-packages (from protobuf>=3.8.0->tensorflow-gpu) (46.3.0)\n","Requirement already satisfied: requests<3,>=2.21.0 in /usr/local/lib/python3.6/dist-packages (from tensorboard<2.3.0,>=2.2.0->tensorflow-gpu) (2.23.0)\n","Requirement already satisfied: werkzeug>=0.11.15 in /usr/local/lib/python3.6/dist-packages (from tensorboard<2.3.0,>=2.2.0->tensorflow-gpu) (1.0.1)\n","Requirement already satisfied: google-auth<2,>=1.6.3 in /usr/local/lib/python3.6/dist-packages (from tensorboard<2.3.0,>=2.2.0->tensorflow-gpu) (1.7.2)\n","Requirement already satisfied: tensorboard-plugin-wit>=1.6.0 in /usr/local/lib/python3.6/dist-packages (from tensorboard<2.3.0,>=2.2.0->tensorflow-gpu) (1.6.0.post3)\n","Requirement already satisfied: google-auth-oauthlib<0.5,>=0.4.1 in /usr/local/lib/python3.6/dist-packages (from tensorboard<2.3.0,>=2.2.0->tensorflow-gpu) (0.4.1)\n","Requirement already satisfied: markdown>=2.6.8 in /usr/local/lib/python3.6/dist-packages (from tensorboard<2.3.0,>=2.2.0->tensorflow-gpu) (3.2.2)\n","Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.6/dist-packages (from requests<3,>=2.21.0->tensorboard<2.3.0,>=2.2.0->tensorflow-gpu) (2020.4.5.1)\n","Requirement already satisfied: idna<3,>=2.5 in /usr/local/lib/python3.6/dist-packages (from requests<3,>=2.21.0->tensorboard<2.3.0,>=2.2.0->tensorflow-gpu) (2.9)\n","Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in /usr/local/lib/python3.6/dist-packages (from requests<3,>=2.21.0->tensorboard<2.3.0,>=2.2.0->tensorflow-gpu) (1.24.3)\n","Requirement already satisfied: chardet<4,>=3.0.2 in /usr/local/lib/python3.6/dist-packages (from requests<3,>=2.21.0->tensorboard<2.3.0,>=2.2.0->tensorflow-gpu) (3.0.4)\n","Requirement already satisfied: pyasn1-modules>=0.2.1 in /usr/local/lib/python3.6/dist-packages (from google-auth<2,>=1.6.3->tensorboard<2.3.0,>=2.2.0->tensorflow-gpu) (0.2.8)\n","Requirement already satisfied: rsa<4.1,>=3.1.4 in /usr/local/lib/python3.6/dist-packages (from google-auth<2,>=1.6.3->tensorboard<2.3.0,>=2.2.0->tensorflow-gpu) (4.0)\n","Requirement already satisfied: cachetools<3.2,>=2.0.0 in /usr/local/lib/python3.6/dist-packages (from google-auth<2,>=1.6.3->tensorboard<2.3.0,>=2.2.0->tensorflow-gpu) (3.1.1)\n","Requirement already satisfied: requests-oauthlib>=0.7.0 in /usr/local/lib/python3.6/dist-packages (from google-auth-oauthlib<0.5,>=0.4.1->tensorboard<2.3.0,>=2.2.0->tensorflow-gpu) (1.3.0)\n","Requirement already satisfied: importlib-metadata; python_version < \"3.8\" in /usr/local/lib/python3.6/dist-packages (from markdown>=2.6.8->tensorboard<2.3.0,>=2.2.0->tensorflow-gpu) (1.6.0)\n","Requirement already satisfied: pyasn1<0.5.0,>=0.4.6 in /usr/local/lib/python3.6/dist-packages (from pyasn1-modules>=0.2.1->google-auth<2,>=1.6.3->tensorboard<2.3.0,>=2.2.0->tensorflow-gpu) (0.4.8)\n","Requirement already satisfied: oauthlib>=3.0.0 in /usr/local/lib/python3.6/dist-packages (from requests-oauthlib>=0.7.0->google-auth-oauthlib<0.5,>=0.4.1->tensorboard<2.3.0,>=2.2.0->tensorflow-gpu) (3.1.0)\n","Requirement already satisfied: zipp>=0.5 in /usr/local/lib/python3.6/dist-packages (from importlib-metadata; python_version < \"3.8\"->markdown>=2.6.8->tensorboard<2.3.0,>=2.2.0->tensorflow-gpu) (3.1.0)\n","Installing collected packages: tensorflow-gpu\n","\u001b[33m  WARNING: The scripts estimator_ckpt_converter, saved_model_cli, tensorboard, tf_upgrade_v2, tflite_convert, toco and toco_from_protos are installed in '/root/.local/bin' which is not on PATH.\n","  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.\u001b[0m\n","Successfully installed tensorflow-gpu-2.2.0\n","Requirement already up-to-date: keras in /usr/local/lib/python3.6/dist-packages (2.3.1)\n","Requirement already satisfied, skipping upgrade: h5py in /usr/local/lib/python3.6/dist-packages (from keras) (2.10.0)\n","Requirement already satisfied, skipping upgrade: six>=1.9.0 in /usr/local/lib/python3.6/dist-packages (from keras) (1.12.0)\n","Requirement already satisfied, skipping upgrade: keras-applications>=1.0.6 in /usr/local/lib/python3.6/dist-packages (from keras) (1.0.8)\n","Requirement already satisfied, skipping upgrade: scipy>=0.14 in /usr/local/lib/python3.6/dist-packages (from keras) (1.4.1)\n","Requirement already satisfied, skipping upgrade: keras-preprocessing>=1.0.5 in /usr/local/lib/python3.6/dist-packages (from keras) (1.1.2)\n","Requirement already satisfied, skipping upgrade: numpy>=1.9.1 in /usr/local/lib/python3.6/dist-packages (from keras) (1.18.4)\n","Requirement already satisfied, skipping upgrade: pyyaml in /usr/local/lib/python3.6/dist-packages (from keras) (3.13)\n","Collecting ann_visualizer\n","  Downloading https://files.pythonhosted.org/packages/db/51/157be500337fba347e32711aaf9f11c1ba9e1162f486a1d708b4ae594ea4/ann_visualizer-2.5.tar.gz\n","Building wheels for collected packages: ann-visualizer\n","  Building wheel for ann-visualizer (setup.py) ... \u001b[?25l\u001b[?25hdone\n","  Created wheel for ann-visualizer: filename=ann_visualizer-2.5-cp36-none-any.whl size=4169 sha256=45b23213944d96ae239c91bd2780d27778af442b9c6b92f44d5e06e8797699ff\n","  Stored in directory: /root/.cache/pip/wheels/b6/b4/4e/d92f50c9c4f004cf315a0e0fcd455486bd799c50fe80cf1f5d\n","Successfully built ann-visualizer\n","Installing collected packages: ann-visualizer\n","Successfully installed ann-visualizer-2.5\n","Requirement already satisfied: graphviz in /usr/local/lib/python3.6/dist-packages (0.10.1)\n"],"name":"stdout"}]},{"cell_type":"markdown","metadata":{"colab_type":"text","id":"J8LEtNbNn7Ri"},"source":["# Training\n","\n"]},{"cell_type":"code","metadata":{"colab_type":"code","id":"A-DO3mGwqq0i","colab":{}},"source":["from datetime import datetime\n","logdir = '/content/drive/My\\ Drive/person_counting/tensorboard/cnn_classification/t2_{}'.format(datetime.now().strftime(\"%Y-%m-%d-%H-%M-%S\"))\n","top_path = '/content/drive/My\\ Drive/person_detection/pcds_dataset_detections/pcds_dataset_detected/'"],"execution_count":0,"outputs":[]},{"cell_type":"code","metadata":{"id":"do7gLVqi_jNB","colab_type":"code","outputId":"f4094eb0-0662-4edc-ca9c-350bce07c4ae","executionInfo":{"status":"ok","timestamp":1589978302425,"user_tz":-120,"elapsed":138329,"user":{"displayName":"horst hammer","photoUrl":"","userId":"16387219538505432860"}},"colab":{"base_uri":"https://localhost:8080/","height":34}},"source":["%cd /content/drive/My\\ Drive/"],"execution_count":8,"outputs":[{"output_type":"stream","text":["/content/drive/My Drive\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"E8ep37lYAEkX","colab_type":"code","outputId":"e8d6a552-4cc2-4d49-82cb-5af29863f50d","executionInfo":{"status":"error","timestamp":1589386004386,"user_tz":-120,"elapsed":158928,"user":{"displayName":"horst hammer","photoUrl":"","userId":"16387219538505432860"}},"colab":{"base_uri":"https://localhost:8080/","height":1000}},"source":["!python -m person_counting.models.cnn_classification --top-path {top_path}\\\n","                                                     --topdir-log {logdir} \\\n","                                                     --n-runs 50 \\\n","                                                     --top-path {top_path}\\\n","                                                     --augmentation-factor 0\\\n","                                                     --filter-hour-above 21\\\n","                                                     --epochs 50\\"],"execution_count":0,"outputs":[{"output_type":"stream","text":["Using TensorFlow backend.\n","/usr/local/lib/python3.6/dist-packages/statsmodels/tools/_testing.py:19: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.\n","  import pandas.util.testing as tm\n","2020-05-20 12:50:44.027782: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcuda.so.1\n","2020-05-20 12:50:44.040687: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2020-05-20 12:50:44.041314: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1561] Found device 0 with properties: \n","pciBusID: 0000:00:04.0 name: Tesla P100-PCIE-16GB computeCapability: 6.0\n","coreClock: 1.3285GHz coreCount: 56 deviceMemorySize: 15.90GiB deviceMemoryBandwidth: 681.88GiB/s\n","2020-05-20 12:50:44.041681: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.1\n","2020-05-20 12:50:44.043158: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10\n","2020-05-20 12:50:44.044854: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcufft.so.10\n","2020-05-20 12:50:44.045236: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcurand.so.10\n","2020-05-20 12:50:44.047256: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusolver.so.10\n","2020-05-20 12:50:44.048174: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusparse.so.10\n","2020-05-20 12:50:44.052100: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudnn.so.7\n","2020-05-20 12:50:44.052238: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2020-05-20 12:50:44.052886: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2020-05-20 12:50:44.053377: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1703] Adding visible gpu devices: 0\n","2020-05-20 12:50:44.053767: I tensorflow/core/platform/cpu_feature_guard.cc:143] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA\n","2020-05-20 12:50:44.059314: I tensorflow/core/platform/profile_utils/cpu_utils.cc:102] CPU Frequency: 2200000000 Hz\n","2020-05-20 12:50:44.059583: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x22aef40 initialized for platform Host (this does not guarantee that XLA will be used). Devices:\n","2020-05-20 12:50:44.059616: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version\n","2020-05-20 12:50:44.153202: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2020-05-20 12:50:44.154008: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x22aed80 initialized for platform CUDA (this does not guarantee that XLA will be used). Devices:\n","2020-05-20 12:50:44.154038: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Tesla P100-PCIE-16GB, Compute Capability 6.0\n","2020-05-20 12:50:44.154225: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2020-05-20 12:50:44.154769: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1561] Found device 0 with properties: \n","pciBusID: 0000:00:04.0 name: Tesla P100-PCIE-16GB computeCapability: 6.0\n","coreClock: 1.3285GHz coreCount: 56 deviceMemorySize: 15.90GiB deviceMemoryBandwidth: 681.88GiB/s\n","2020-05-20 12:50:44.154846: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.1\n","2020-05-20 12:50:44.154866: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10\n","2020-05-20 12:50:44.154882: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcufft.so.10\n","2020-05-20 12:50:44.154897: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcurand.so.10\n","2020-05-20 12:50:44.154911: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusolver.so.10\n","2020-05-20 12:50:44.154925: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusparse.so.10\n","2020-05-20 12:50:44.154940: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudnn.so.7\n","2020-05-20 12:50:44.155009: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2020-05-20 12:50:44.155537: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2020-05-20 12:50:44.156015: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1703] Adding visible gpu devices: 0\n","2020-05-20 12:50:44.156079: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.1\n","2020-05-20 12:50:44.157278: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1102] Device interconnect StreamExecutor with strength 1 edge matrix:\n","2020-05-20 12:50:44.157307: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1108]      0 \n","2020-05-20 12:50:44.157315: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1121] 0:   N \n","2020-05-20 12:50:44.157428: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2020-05-20 12:50:44.158013: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2020-05-20 12:50:44.158589: W tensorflow/core/common_runtime/gpu/gpu_bfc_allocator.cc:39] Overriding allow_growth setting because the TF_FORCE_GPU_ALLOW_GROWTH environment variable is set. Original config value was 0.\n","2020-05-20 12:50:44.158639: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1247] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 15056 MB memory) -> physical GPU (device: 0, name: Tesla P100-PCIE-16GB, pci bus id: 0000:00:04.0, compute capability: 6.0)\n","Dataset contains: \n","788 training csvs \n","265 testing csvs\n","Model: \"model_1\"\n","_________________________________________________________________\n","Layer (type)                 Output Shape              Param #   \n","=================================================================\n","input_1 (InputLayer)         (None, 573, 47, 1)        0         \n","_________________________________________________________________\n","average_pooling2d_1 (Average (None, 286, 23, 1)        0         \n","_________________________________________________________________\n","conv2d_1 (Conv2D)            (None, 286, 23, 4)        40        \n","_________________________________________________________________\n","average_pooling2d_2 (Average (None, 143, 11, 4)        0         \n","_________________________________________________________________\n","conv2d_2 (Conv2D)            (None, 143, 11, 4)        148       \n","_________________________________________________________________\n","lambda_1 (Lambda)            (None, 143, 44)           0         \n","_________________________________________________________________\n","lstm_1 (LSTM)                (None, 10)                2200      \n","_________________________________________________________________\n","dense_1 (Dense)              (None, 10)                110       \n","=================================================================\n","Total params: 2,498\n","Trainable params: 2,498\n","Non-trainable params: 0\n","_________________________________________________________________\n","Actual model is using following hyper-parameters:\n","units :  5\n","regularization :  0.0\n","pooling_type :  avg\n","pool_size_y :  2\n","pool_size_x :  2\n","optimizer :  Adam\n","loss :  categorical_crossentropy\n","learning_rate :  0.00023418306436418312\n","layer_number :  2\n","kernel_size :  3\n","kernel_number :  4\n","filter_rows_lower :  77\n","filter_rows_factor :  1\n","filter_cols_upper :  15\n","filter_cols_lower :  18\n","batch_size :  128\n","batch_normalization :  False\n","filter_hour_above :  21\n","filter_category_noisy :  False\n","Generator contains  265  files with  2  batches of size  128\n","Generator contains  788  files with  6  batches of size  128\n","/usr/local/lib/python3.6/dist-packages/keras/engine/training_generator.py:49: UserWarning: Using a generator with `use_multiprocessing=True` and multiple workers may duplicate your data. Please consider using the `keras.utils.Sequence class.\n","  UserWarning('Using a generator with `use_multiprocessing=True`'\n","Epoch 1/50\n","/usr/local/lib/python3.6/dist-packages/keras/utils/data_utils.py:718: UserWarning: An input could not be retrieved. It could be because a worker has died.We do not have any information on the lost sample.\n","  UserWarning)\n","2020-05-20 12:51:40.377096: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10\n","2020-05-20 12:51:41.837212: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudnn.so.7\n","5/6 [========================>.....] - ETA: 11s - loss: 2.3021 - f1: 0.0000e+00 - categorical_accuracy: 0.2313"],"name":"stdout"}]},{"cell_type":"markdown","metadata":{"id":"Fwo7obY7DcUl","colab_type":"text"},"source":[""]},{"cell_type":"markdown","metadata":{"id":"NQFwhx_FWzGV","colab_type":"text"},"source":["# Testing for best params\n","\n","### Set static params in test file"]},{"cell_type":"code","metadata":{"id":"I2bdEgA9VZUp","colab_type":"code","colab":{"base_uri":"https://localhost:8080/","height":383},"outputId":"94260c41-c398-499e-aad6-5de047d6579e","executionInfo":{"status":"ok","timestamp":1589731855025,"user_tz":-120,"elapsed":126185,"user":{"displayName":"horst hammer","photoUrl":"","userId":"16387219538505432860"}}},"source":["from pandas.plotting import register_matplotlib_converters\n","import seaborn as sns\n","from matplotlib import rc\n","from pylab import rcParams\n","\n","import matplotlib.pyplot as plt\n","%matplotlib inline\n","\n","plt.style.use('classic')\n","\n","!python -m person_counting.tests.test_lstm train_best_gpu"],"execution_count":7,"outputs":[{"output_type":"stream","text":["/usr/local/lib/python3.6/dist-packages/statsmodels/tools/_testing.py:19: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.\n","  import pandas.util.testing as tm\n"],"name":"stderr"},{"output_type":"stream","text":["/usr/local/lib/python3.6/dist-packages/statsmodels/tools/_testing.py:19: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.\n","  import pandas.util.testing as tm\n","Using TensorFlow backend.\n","Traceback (most recent call last):\n","  File \"/usr/lib/python3.6/runpy.py\", line 193, in _run_module_as_main\n","    \"__main__\", mod_spec)\n","  File \"/usr/lib/python3.6/runpy.py\", line 85, in _run_code\n","    exec(code, run_globals)\n","  File \"/content/drive/My Drive/person_counting/tests/test_lstm.py\", line 18, in <module>\n","    from person_counting.tests.test_cnn import test_input_csvs\n","  File \"/content/drive/My Drive/person_counting/tests/test_cnn.py\", line 15, in <module>\n","    from person_counting.models import cnn_classification as cnn_cls\n","  File \"/content/drive/My Drive/person_counting/models/cnn_classification.py\", line 20, in <module>\n","    from person_counting.bin.evaluate_cls import evaluate_run_cls, f1\n","  File \"/content/drive/My Drive/person_counting/bin/evaluate_cls.py\", line 111\n","    video_type.append(get_video_class(file_name, df_y).values[0]])\n","                                                                ^\n","SyntaxError: invalid syntax\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"kOfBHA8FTVdB","colab_type":"code","colab":{}},"source":["# use this code to prevent colab from disconnecting during longer training times in javascript console window\n","function ClickConnect(){\n","    console.log(\"Clicked on connect button\"); \n","    document.querySelector(\"colab-connect-button\").click()\n","}\n","setInterval(ClickConnect,60000)"],"execution_count":0,"outputs":[]}]}